dali/image/image_factory.cc

```diff
@@ -92,7 +96,11 @@ ImageFactory::CreateImage(const uint8_t *encoded_image, size_t length, DALIImage
   } else if (CheckIsGIF(encoded_image, length)) {
     DALI_FAIL("GIF format is not supported");
   } else if (CheckIsTiff(encoded_image, length)) {
+#ifdef DALI_USE_LIBTIFF
+    return std::make_unique<TiffImage_LibtiffImpl>(encoded_image, length, image_type);
+#else
```

Comment:
I'd prefer simple `if` here. `ifdefs` always turn out to be pain in the ass :(

Reply:
it's done now

Reply:
As we don't have the sources for the library at all in that scenario, I think that #if is mandated.